### PR TITLE
Shaka: manually fetch segments and control playback stream level

### PR DESCRIPTION
**Pull request by i-zolotarenko**

### CODE REVIEW

**Comment by ChronoBrake** [comment]
wow you come back! awesome! i love Shaka Player with P2P loader, wait for support latest shaka player version!
Good Luck!!

**Comment by ChronoBrake** [comment]
Any ETA when update will be realesed? @i-zolotarenko @mrlika

**Comment by mrlika** [comment]
> Any ETA when update will be realesed? @i-zolotarenko @mrlika

Unfortunately no ETA. Hopefully soon.

File: packages/p2p-media-loader-shaka/src/services/loading-handler.ts [hunk 1]
```diff
@@ -1,45 +1,129 @@
...
+      response.timeMs = getLoadingDurationBasedOnBitrate({
+        bitrate: 2749539,
+        bytesLoaded: response.data.byteLength,
+      });
+
+      return response;
+    });
+    return new this.shaka.util.AbortableOperation(
+      promise,
+      async () => undefined
```
Comment (mrlika) [inline packages/p2p-media-loader-shaka/src/services/loading-handler.ts:92]: Add abort logic

File: packages/p2p-media-loader-shaka/src/services/loading-handler.ts [hunk 2]
```diff
@@ -1,45 +1,129 @@
...
+    this.debug(`Stream id: ${stream?.id}`);
+    this.debug(`Segment: ${segment?.index}`);
+    if (!stream) return this.defaultLoad();
+
+    const promise = this.fetchSegment(segmentUrl, byteRangeString);
+    promise.then((response) => {
+      response.timeMs = getLoadingDurationBasedOnBitrate({
+        bitrate: 2749539,
+        bytesLoaded: response.data.byteLength,
+      });
```
Comment (mrlika) [inline packages/p2p-media-loader-shaka/src/services/loading-handler.ts:86]: Try move to `fetchSegment`

**Review by mrlika — APPROVED** [review]

**Pull request merged**